### PR TITLE
chore(docker): containerize repo

### PR DESCRIPTION
**Pull request by chloelbn**
| Q                 | A
| ----------------- | ----------
| Bug fix?          | no
| New feature?      | no
| BC breaks?        | no 

## Describe your change

Add a `Dockerfile` to allow the use of the repo in a [Docker](https://www.docker.com/) container

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -112,6 +112,11 @@ For full documentation, visit the **[Algolia Symfony Search Bundle](https://www.
 
 Encountering an issue? Before reaching out to support, we recommend heading to our [FAQ](https://www.algolia.com/doc/framework-integration/symfony/troubleshooting/faq/) where you will find answers for the most common issues and gotchas with the bundle.
 
+## Use the Dockerfile
+
+If you wish to contribute to the repository but would like to avoid installing the dependencies locally, we provided you with a Docker image.
+Please check our [dedicated guide](DOCKER_README.MD) to know more.
+
```
Comment (DevinCodes) [inline README.md:119]:
```suggestion
If you want to contribute to this project without installing all its dependencies, you can use our Docker image. Please check our [dedicated guide](DOCKER_README.MD) to learn more.
```

**Review by DevinCodes — CHANGES_REQUESTED** [review]

File: DOCKER_README.MD [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+
+```bash
+docker run -it --rm --env ALGOLIA_APP_ID=XXXXXX [...] -v $PWD:/app -w /app algolia-search-bundle bash
+```
+
+But we advise you export them in your `.bashrc` or `.zshrc`. That way, you can use [Docker's shorten syntax](https://docs.docker.com/engine/reference/commandline/run/#set-environment-variables--e---env---env-file) to retrieve your variables.
+
+```bash
+### For external contributors, the following env variables should be enough
+docker run -it --rm --env ALGOLIA_APP_ID \
```
Comment (DevinCodes) [inline DOCKER_README.MD:56]: I ran this command after the setup, but it didn't work; I had to manually run `composer install` before it worked. This was my output:

```
composer test
Do not run Composer as root/super user! See https://getcomposer.org/root for details
> php-cs-fixer fix -v --dry-run
sh: 1: php-cs-fixer: not found
Script php-cs-fixer fix -v --dry-run handling the test:lint event returned with error code 127
Script @test:lint was called via test
```

It looks like the `RUN composer install` in the `Dockerfile` doesn't do as expected. Any idea why?

**Review by DevinCodes — CHANGES_REQUESTED** [review]

File: DOCKER_README.MD (hunk 2 above)
Comment (chloelbn) [reply DOCKER_README.MD:55]: @DevinCodes could you try again? it may related to the fact I didn't put the `vendor/` folder in the `.dockerignore` file

Comment (DevinCodes) [reply DOCKER_README.MD:55]: I still have the same issue :(

**Comment by DevinCodes** [comment]
Can you rebase `master` on this branch, please? I think that's why the CI is failing 😕

**Comment by chloelbn** [comment]
@DevinCodes I'll rebase after the other PR will be merged!

File: DOCKER_README.MD [hunk 3]
```diff
@@ -0,0 +1,79 @@
+In this page you will find our recommended way of installing Docker on your machine. 
+This guide is made for OSX users.
+
+## Install docker
```
Comment (DevinCodes) [inline DOCKER_README.MD:4]: Can you apply the suggested changes in the other repos on this README, please? E.g. `Docker` is not capitalized everywhere in this file. Thank you 😄

**Review by DevinCodes — CHANGES_REQUESTED** [review]

File: DOCKER_README.MD [hunk 4]
```diff
@@ -0,0 +1,79 @@
...
+```
+
+Then install [VirtualBox](https://www.virtualbox.org/) with [Homebrew Cask](https://github.com/Homebrew/homebrew-cask) to get a driver for your Docker machine
+```
+$ brew cask install virtualbox
+```
+
+You may need to enter your password and authorize the application in your `System Settings` > `Security & Privacy`.
+
+Create now a new machine, set it up as default and connect your shell to it (here we use zsh. The commands should anyway be displayed in each steps' output)
```
Comment (DevinCodes) [inline DOCKER_README.MD:27]:
```suggestion
Create a new machine, set it up as default and connect your shell to it.
```

File: DOCKER_README.MD [hunk 5]
```diff
@@ -0,0 +1,79 @@
...
+
+```
+$ docker-machine create --driver virtualbox default
+$ docker-machine env default
+$ eval "$(docker-machine env default)"
+```
+
+Now you're all setup to use our provided Docker image!
```
Comment (DevinCodes) [inline DOCKER_README.MD:35]:
```suggestion
Now you're all set to use our Docker image!
```

File: DOCKER_README.MD [hunk 6]
```diff
@@ -0,0 +1,79 @@
...
+
+## Build the image
+
+```bash
+docker build -t algolia-search-bundle .
+```
+
+## Run the image
+
+You need to provide few environment variables at runtime to be able to run the [Common Test Suite](https://github.com/algolia/algoliasearch-client-specs/tree/master/common-test-suite).
```
Comment (DevinCodes) [inline DOCKER_README.MD:45]:
```suggestion
You need to provide a few environment variables at runtime in order to run the [Common Test Suite](https://github.com/algolia/algoliasearch-client-specs/tree/master/common-test-suite).
```

File: DOCKER_README.MD [hunk 7]
```diff
@@ -0,0 +1,79 @@
...
+
+```bash
+docker run -it --rm --env ALGOLIA_APP_ID \
+                    --env ALGOLIA_API_KEY \
+-v $PWD:/app -w /app algolia-search-bundle bash
+```
+
+Once your container is running, any changes you make in your IDE are directly reflected in the container.
+
+Manually install composer dependencies once inside the container
```
Comment (DevinCodes) [inline DOCKER_README.MD:62]:
```suggestion
The first time you start the container, you have to manually install the composer dependencies.
```

**Review by DevinCodes — CHANGES_REQUESTED** [review]
Some small feedback, after that it's good to go I think 🙂

**Review by DevinCodes — APPROVED** [review]

**Review by DevinCodes — CHANGES_REQUESTED** [review]
Ah sorry, approved by accident, I see the CI is still failing for some jobs

**Comment by chloelbn** [comment]
@DevinCodes yes, for some reason those two images use Xdebug 3.0.0 instead of 2.9/2.8 and it crashes. I'm investigating why

**Review by DevinCodes — APPROVED** [review]

**Pull request merged**